Add a recipe for bytecode package

MatthieuDartiailh opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in 
a comment on the PR to get the attention of the `staged-recipes` team. You can 
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
if your recipe isn't reviewed promptly.
-->

Bytecode is Python bytecode assembler disassembler created initially by vstinner. It will be used in the next version of enaml for which a feedstock does exist. 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/bytecode```) and found some lint.

Here's what I've got...


For **recipes/bytecode**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [27]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/bytecode```) and found it was in an excellent condition.

npavlovikj commented on recipes/bytecode/meta.yaml:
```diff
@@ -0,0 +1,58 @@
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  skip: True  # [py<35]
+  script: "{{ PYTHON }} -m pip install . -vv"
```
I think this recipe can be `noarch`, https://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages. Thus, please remove the `skip` line and add `noarch: python` instead.

npavlovikj commented on recipes/bytecode/meta.yaml:
```diff
@@ -0,0 +1,58 @@
+
+requirements:
+  host:
+    - python
```
Use `python >=3.5`.

npavlovikj commented on recipes/bytecode/meta.yaml:
```diff
@@ -0,0 +1,58 @@
+    - python
+    - pip
+  run:
+    - python
```
Use `python >=3.5`.

npavlovikj commented on recipes/bytecode/meta.yaml:
```diff
@@ -0,0 +1,58 @@
+    - pip
+  run:
+    - python
+    - aenum  # [py<36]
```
Please try removing the selector, `# [py<36]`.

MatthieuDartiailh replied:
aenum is not required on python 3.6 and above which is why I used the selector. Should we install it unconditionally even if it is not used ?

npavlovikj replied:
Ah, I see. Yeah, if it is not needed at all, you can remove it. I think CF builds against Python >=3.6 anyway.

MatthieuDartiailh replied:
I removed aeunm and in consequence bumped the minimum python version to 3.6. I could not confirm that CF only build on python 3.6 and above so it sounds safer to make sure we do not generate a bad package.

npavlovikj commented:
@MatthieuDartiailh , the AppVeyor error can be ignored. I added few comments, so please address them. Moreover, please clean the recipe and remove all comments and extra line in the `about` section.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/bytecode```) and found some lint.

Here's what I've got...


For **recipes/bytecode**:

 * license_file entry is missing, but is required.
 * ``requirements: host: python >= 3.5`` should not contain a space between relational operator and the version, i.e. ``python >=3.5``
 * ``requirements: run: python >= 3.5`` should not contain a space between relational operator and the version, i.e. ``python >=3.5``

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/bytecode```) and found some lint.

Here's what I've got...


For **recipes/bytecode**:

 * license_file entry is missing, but is required.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/bytecode```) and found it was in an excellent condition.

MatthieuDartiailh commented:
@npavlovikj I addressed all your comments even though I am not convinced this package should be noarch as explained above. Thanks for the quick review BTW.

npavlovikj commented on recipes/bytecode/meta.yaml:
```diff
@@ -0,0 +1,42 @@
+  sha256: f78a312880173f1d20aef2eceb8eb959dc6c2167966b757fef201b8af839f802
+
+build:
+  noarch: True
```
This should be `noarch: python`.

npavlovikj commented on recipes/bytecode/meta.yaml:
```diff
@@ -0,0 +1,42 @@
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
```
Please remove these comments as well.

MatthieuDartiailh commented:
ping @npavlovikj

npavlovikj commented:
The recipe looks good @MatthieuDartiailh , thank you for contributing 🎉

npavlovikj commented:
> @npavlovikj I addressed all your comments even though I am not convinced this package should be noarch as explained above. Thanks for the quick review BTW.

In general, if a recipe doesn't require compilers or architecture dependent packages, it can be `noarch`, https://conda-forge.readthedocs.io/en/latest/meta.html#building-noarch-packages.

MatthieuDartiailh commented:
Thanks @npavlovikj
